File: Cakeminator.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    int n, m;
    cin >> n >> m;
    set<int> col, row;
    for (int i=0; i<n; i++) {
        for (int j=0; j<m; j++) {
            char c;
            cin >> c;
            if (c=='S') {
                row.insert(i);
                col.insert(j);
            }
        }
    }
    int ans = n*m - col.size()*row.size();
    cout << ans << endl;

    return 0;
}

File: Python/Cakeminator.py
def main():
    n, m = map(int, input().split())
    col = set()
    row = set()
    for i in range(n):
        for j in range(m):
            c = input()
            if c == "S":
                row.add(i)
                col.add(j)

    ans = n*m-len(col)*len(row)
    print(ans)


if __name__ == '__main__':
    main()
